hoodie-common/src/main/java/com/uber/hoodie/common/model/HoodieRecordPayload.java

```diff
@@ -56,4 +57,10 @@
      * Return EMPTY to skip writing this record.
      */
     Optional<IndexedRecord> getInsertValue(Schema schema) throws IOException;
+
+    /**
+     * This method can be used to fetch metadata regarding the HoodieRecordPayload, for the purpose
+     * of capturing some metrics.
+     */
+    Map<String, String> getMetadata();
```

Comment:
In fact, can you make this an `Optional<Map<String, String>>` and always return Optional.empty from the base class?